# <center> hw1

In [1]:
import numpy as np

In [22]:
array1=np.genfromtxt('homes.csv',skip_header=1,delimiter=',')
array1=array1[:,0:8]
array1.shape

(522, 8)

In [23]:
x,y=array1[:,2],array1[:,0]
x1=np.c_[np.ones(522),x]

In [24]:
b1=np.dot(x1.T,x1)
b1=np.linalg.inv(b1)
b2=np.dot(x1.T,y)
coeffs=np.dot(b1,b2).reshape(-1,1)
coeffs

array([[82808.79961262],
       [56200.08366568]])

The estimated increase in the average price for each additional bedroom is 56200.08 dollars

In [28]:
xn=array1[:,[1,2,3,4,7]]
xn=np.c_[np.ones(522),xn]

In [29]:
b1=np.dot(xn.T,xn)
b1=np.linalg.inv(b1)
b2=np.dot(xn.T,y)
coeffs=np.dot(b1,b2).reshape(-1,1)
coeffs

array([[-1.13113307e+05],
       [ 1.28056651e+02],
       [-1.47934409e+04],
       [ 1.96186361e+04],
       [ 3.64244996e+04],
       [ 1.00804986e+00]])

In [39]:
newval=np.median(xn,axis=0)
newval=newval.reshape(1,6)
newval

array([[1.000e+00, 2.061e+03, 3.000e+00, 3.000e+00, 2.000e+00, 2.220e+04]])

In [40]:
np.dot(newval,coeffs)

array([[260514.74311249]])

The price is estimated to be 260514.74 dollars.